src/npm/package/package.ts

```diff
@@ -10,80 +10,18 @@ import {
   APIGatewayProxyResult,
   Context,
 } from "aws-lambda";
-import { DocumentClient } from "aws-sdk/clients/dynamodb";
-import { getRegistryEntryForPackage } from "../util/registry";
 import "source-map-support/register";
-import * as zlib from "zlib";
-
-const DDBDocClient = new DocumentClient();
-const tableName = process.env.DDB_TABLE!;
-
-/** Get or create DDB entry with map of files for package and return it
- *
- * Registry data is stored compressed in DDB for 2 reasons:
- * 1) Not all registry data maps cleanly to a DDB item (e.g. empty strings)
- * 2) Some items (e.g. @types/node) have too much data to store uncompressed
- */
-async function getPackageJson(
-  packageName: string,
-  downloadPrefix: string,
-): Promise<string> {
-  try {
-    console.log(
-      "Checking " + tableName + " for existing " + packageName + " entry.",
-    );
-    const cachedList = await DDBDocClient.get({
-      TableName: tableName,
-      Key: { PackageName: packageName },
-    }).promise();
-    console.log("Existing package found; returning it.");
-    return zlib.inflateSync(cachedList.Item!.CompressedRegistryData).toString();
-  } catch (ddbGetError) {
-    console.log(
-      "No existing " +
-        packageName +
-        " entry found; retrieving it from upstream.",
-    );
-    var registryData = await getRegistryEntryForPackage(packageName);
-
-    // update download links to point to proxy endpoint
-    Object.keys(registryData["versions"]).forEach(function(
-      value,
-      _index,
-      _array,
-    ) {
-      registryData["versions"][value]["dist"]["tarball"] =
-        downloadPrefix + value;
-    });
-
-    await DDBDocClient.put({
-      TableName: tableName,
-      Item: {
-        PackageName: packageName,
-        CompressedRegistryData: zlib.deflateSync(JSON.stringify(registryData)),
-      },
-    }).promise();
-    return JSON.stringify(registryData);
-  }
-}
+import { app } from "../NpmApp";
 
 /** AWS Lambda entrypoint */
 export let handler: APIGatewayProxyHandler = async (
   event: APIGatewayProxyEvent,
   context: Context,
 ): Promise<APIGatewayProxyResult> => {
-  var npmPackageName = decodeURIComponent(event!.pathParameters!.proxy);
-  var downloadUriPrefix =
-    "https://" +
-    event!.requestContext.domainName +
-    "/" +
-    event!.requestContext.stage +
-    "/npm-dlredirect/" +
-    npmPackageName +
-    "/";
-
+  const npmPackageName = decodeURIComponent(event!.pathParameters!.proxy);
+  const npmPackage = app.getNpmPackage(npmPackageName);
```

Comment:
I realize now that some of the coverage issues here are totally my fault -- I thought this project already had more coverage, specifically including the handlers, but I realize now I was getting it confused with another project (e.g. [handler](https://github.com/troyready/git-lfs-s3/blob/master/src/batch/batch.ts#L135-L180) / [test](https://github.com/troyready/git-lfs-s3/blob/master/src/batch/batch.test.ts)).

So it's not a new problem now, but I have some concern about the gap in testing around the handler and NpmApp class itselff. I'm open to your suggestions on a path forward (I'm a little too narrow focused at the moment on catching every code path so I'd like to focus more testing at the entry point), but as I said in another comment that doesn't need to be in this PR.

Reply:
I refactored and added tests for the handlers involved in the scope of this PR.
fixed in https://github.com/troyready/serverless-artifact-cacher/pull/3/commits/672162b433d3a7f766e7909e125a20489cf9822a